app/controllers/dashboard/form/members_controller.rb

```diff
@@ -4,7 +4,9 @@ module Dashboard
   module Form
     class MembersController < BaseController
       def edit
-        @resource = Collection.find(params[:id])
+        @resource = Collection
+          .includes(collection_work_memberships: [:work])
+          .find(params[:id])
```

Comment:
Not the cause of the bug, but I figured while I was in here looking around I would boost the db performance a little.